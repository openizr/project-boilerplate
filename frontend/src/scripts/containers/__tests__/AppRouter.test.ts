/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/vue';
import AppRouter from 'scripts/containers/AppRouter.vue';

jest.mock('scripts/store/routes', () => ({
  '/': (): Any => Promise.resolve(require('scripts/pages/HomePage.vue').default),
}));

jest.mock('scripts/store/index', () => ({
  mutate: jest.fn(),
  dispatch: jest.fn(),
  useCombiner: jest.fn((_hash, callback) => callback({ route: (process.env.NOT_FOUND === 'true' ? null : '/') })),
}));

describe('containers/AppRouter', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly - loading page', () => {
    const { container } = render(AppRouter, { props: { locale: { LABEL_TEST: 'Test' } } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - page found', async () => {
    const { container } = render(AppRouter, { props: { locale: { LABEL_TEST: 'Test' } } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - page not found', async () => {
    process.env.NOT_FOUND = 'true';
    const { container } = render(AppRouter, { props: { locale: { LABEL_TEST: 'Test' } } });
    expect(container.firstChild).toMatchSnapshot();
  });
});
