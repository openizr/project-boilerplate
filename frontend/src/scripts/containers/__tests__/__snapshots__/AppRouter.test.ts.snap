// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/AppRouter renders correctly - loading page 1`] = `
<div
  locale="[object Object]"
>
  Loading...
</div>
`;

exports[`containers/AppRouter renders correctly - page found 1`] = `
<div
  locale="[object Object]"
>
  Loading...
</div>
`;

exports[`containers/AppRouter renders correctly - page not found 1`] = `<!--v-if-->`;
